vllm/core/scheduler.py

```diff
@@ -410,6 +514,18 @@ def __init__(
         # for processing and deallocation by the free_finished_seq_groups()
         self._async_stopped: List[SequenceGroup] = []
 
+        # List with the chunk sizes to hand out to each sequence depending
+        # on how many partial prefills are running. This is slightly faster than
+        # running an integer division every time a prefill is scheduled.
+        # This splits the budget evenly among all prefills.
+        self.partial_prefill_budget_lookup_list = [0] * (
+            self.scheduler_config.max_num_partial_prefills + 1)
+        self.partial_prefill_budget_lookup_list[0] = (
+            scheduler_config.max_num_batched_tokens)
```

Comment:
> partial_prefill_budget_lookup_list is a list where num partial prefills -> token budget for each partial prefill?

Yup! We calculate this once up-front when the scheduler is created so that we don't have to do a division for every sequence in every scheduling step. A list access measures slightly faster than an integer division, at least on the machines we have running in IBM cloud.

 > If we want to keep this, I think moving this into the metadata class might be better?

Yeah we could do that, I just wasn't quite sure how to make that properly static and cached on the metadata class, since we wouldn't want to re-create this list on every scheduler step

Reply:
I spent a few minutes looking into this and I'm not sure there's a clean way to add a class attribute to a @dataclass, unfortunately. I'd rather not make the code even more hacky than this is!

What do you think between either:
1. Leaving as is, or
2. Doing the division in-line like `scheduler_config.max_num_batched_tokens // partial_prefill_metadata.schedulable_prefills` ?